e2e/test/scenarios/joins/reproductions/23293-drill-through-implicit-join-add-remove-column.cy.spec.js

```diff
@@ -0,0 +1,96 @@
+import {
+  restore,
+  popover,
+  openOrdersTable,
+  visitDashboard,
+} from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { ORDERS, PRODUCTS } = SAMPLE_DATABASE;
+
+describe("issue 23293", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should retain the filter when drilling through the dashboard card with implicitly added column (metabase#23293)", () => {
+    openOrdersTable();
+
+    cy.findByTestId("viz-settings-button").click();
+    modifyColumn("Product ID", "remove");
+    modifyColumn("Category", "add");
+    cy.wait("@dataset");
+
+    saveQuestion().then(
+      ({
+        response: {
+          body: { id },
+        },
+      }) => {
+        const questionDetails = {
+          query: {
+            "source-table": `card__${id}`,
+            aggregation: [["count"]],
+            breakout: [
+              [
+                "field",
+                PRODUCTS.CATEGORY,
+                {
+                  "source-field": ORDERS.PRODUCT_ID,
+                },
+              ],
+            ],
+          },
+          display: "bar",
+        };
+
+        cy.createQuestionAndDashboard({ questionDetails }).then(
+          ({ body: { dashboard_id } }) => {
+            visitDashboard(dashboard_id);
+          },
+        );
+
+        cy.get(".bar").first().realClick();
+        popover()
+          .findByText(/^See these/)
+          .click();
+
+        cy.findByTestId("qb-filters-panel").should(
+          "contain",
+          "Product → Category is Doohickey",
+        );
+        cy.findAllByTestId("header-cell")
+          .last()
+          .should("have.text", "Product → Category");
+
+        cy.findAllByRole("grid")
+          .last()
+          .as("tableResults")
+          .should("contain", "Doohickey")
+          .and("not.contain", "Gizmo");
+      },
+    );
+  });
+});
+
+/**
+ * @param {string} columnName
+ * @param {("add"|"remove")} action
+ */
+function modifyColumn(columnName, action) {
+  const icon = action === "add" ? "add" : "eye_outline";
+  const iconSelector = `.Icon-${icon}`;
+  const columnSeletor = `draggable-item-${columnName}`;
+  cy.findByTestId(columnSeletor).find(iconSelector).click();
+}
+
+function saveQuestion() {
+  cy.intercept("POST", "/api/card").as("saveQuestion");
+
+  cy.findByTestId("qb-header-action-panel").findByText("Save").click();
+  cy.get(".Modal").button("Save").click();
```

Comment:
This is one of the examples I said in the comment above. If you look at the implementation
https://github.com/metabase/metabase/blob/e5b4b8a5d977860499b0c73630161f28e7a24124/e2e/support/commands/ui/button.js#L2

It doesn't take the previous subjects into account, so this behaves just like `cy.button`. Can you change this to use `.within` for now just to make sure it does what you think it does?

```suggestion
  cy.get(".Modal").within(() => {
    cy.button("Save").click();
  });```

Reply:
Thanks for being so vigilant.
I am aware of this and I've previously prepared a PR that addresses this problem for the `icon` custom command. The next step is to do the same for the `button`.

If you don't mind, I'd rather leave the test in this format so that I don't have to remove `.within()` after the button refactor?

Reply:
Sounds good to me 👍